### ISSUE TITLE
alias/relabel with {id -> string} mapping

### ISSUE THREAD

**Issue by oliveagle** [VictoriaMetrics/VictoriaMetrics#369]
**Is your feature request related to a problem? Please describe.**

In many case  some label values are all numbers, such as service code, error code, xxx code.  such numbers is very difficult for reading if we group by that label.  

```
some_metric{ code = '001' }  1
some_metric{ code = '002' }  1
```
we want to group by `code`

```
sum(some_metric) by (code)
```

for more human readable, we need to relabel it's value(or alias).  we can use these two query to do it.
```
alias(sum(some_metric{code = '001'}, 'MakeCoffee')
alias(sum(some_metric{code = '002'}, 'Drink')
```

but what if we have 100 or more of codes? we have to use many queries with separately on each code.

**Describe the solution you'd like**

can we have a mapping function to work around ? e.g.:

label_mapping(q, label,  key1, value1, key2, value2, key3, value3, ...keyN, valueN)

which will do some thing like this:

```
func label_mapping(q, label, key1, value1, key2, value2, key3, value3, ...keyN, valueN):
    tmp_dict = {key1: value1, key2: value2, key3: value3, ..., keyN: valueN}
    v = get_label_value_from_q(q, label)
    return label_replace(q, label, tmp_dict[v])
```

then we can use one single query to replace or set a label value for human

```
label_mapping(
    sum(some_metric) by (code), 
    '001', 'MakeCoffee', 
    '002', 'Drink'
)
```

**Comment by valyala** [comment]
@oliveagle , thanks for the feature request. The `label_map(q, label, srcValue1, dstValue1 [, ... srcValueN, dstValueN])` function has been added to MetricsQL in the commit fdc2a9d1d72fb24796eec33675bf431a79afb218 . This commit will be included in the next release. In the meantime you can build VictoriaMetrics from this commit according to [these instructions](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#how-to-build-from-sources) and verify whether the `label_map` function works as expected.

**Comment by valyala** [comment]
The handling of `label_map()` has been improved in the commit 3c7c71a49cd8b61efc933ea53cef828128a9f7b0 . Now it doesn't touch the label value if there is no matching `srcValue` in the list passed to `label_map()`. Additionally, it can be used for the following cases:
* Adding missing label: `label_map(q, "label", "", "foo")` would add `label="foo"` to all the time series returned by `q` with missing `label`.
* Removing label with the given value: `label_map(q, "label", "foo", "")` would remove `label="foo"`` from all the time series returned by `q`.

**Comment by valyala** [comment]
@oliveagle , the `label_map()` function is available starting from [v1.34.4](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
